### PR TITLE
[Part 6] Introduce a temp table mode

### PR DESCRIPTION
**Pull request by kamal**
Currently, directory diffing works by loading both directories in memory and performing a line-by-line comparison. For small directories, this works well, but we risk running out of memory for very large directories.

This PR adds an option to perform the diffing using SQL. This works by loading the current directory and the incoming csv as temp tables and performing a few queries in order to identify new records, deleted records and changed records.

~~One big difference in the implementation with the in memory version is we bunch up all the `inserts` first, then `deletes`, followed by `updates` and `noop`. The idea is that any record inside these blocks of operations don't really have strict ordering between the rows. In the in-memory implementation, we traverse the records depth-first, so that dependent assistant records are processed before their owners. To smooth over these differences, I introduced an `Operations` class that overrides `==` so that specs pass in either version.~~

Another difference is the expected input for the temp table mode. The in-memory mode expects an array of arrays of the form `[name, email, phone_number, assistants]`. The temp table expects an `ActiveRecord` relation for both the current and CSV. For the CSV, we can use something like https://github.com/mynameisrufus/theman to load into a table.

### CODE REVIEW

**Comment by kamal** [comment]
I'm going to work on getting the output from temp table to match the csv without the Operations hackery. Design-wise, it feels like I should `insert as select` for each operation while keeping the original csv row number. Then, `select *` all rows ordered by row number, and perform light reordering based on assistant field.

**Comment by kamal** [comment]
> I'm going to work on getting the output from temp table to match the csv without the Operations hackery. Design-wise, it feels like I should `insert as select` for each operation while keeping the original csv row number. Then, `select *` all rows ordered by row number, and perform light reordering based on assistant field.

This is done.

File: lib/directory_diff/transformer/temp_table.rb [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+      def add_operation(operation)
+        op = [
+          operation.operation.to_sym,
+          operation.name,
+          operation.email,
+          operation.phone_number.presence,
+          serialize_pg_array(operation.assistants)
+        ]
+        op << operation.extra unless operation[:extra].nil?
```
Comment (abuiles) [inline lib/directory_diff/transformer/temp_table.rb]: @kamal what is this for?

Comment (kamal) [reply lib/directory_diff/transformer/temp_table.rb]: If the CSV has an extra column after assistant, we serialize it out. One of the tests show this behavior with the extra field `'foo'`

**Pull request merged**